src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCreationService.java

```diff
@@ -88,6 +115,105 @@ public void bulkCreationLogic(String userToken, MultipleDetails multipleDetails,
 
     }
 
+    private void addLegalRepsFromSinglesCases(MultipleDetails multipleDetails) throws IOException {
+        MultipleData multipleData = multipleDetails.getCaseData();
+        String leadCaseId = multipleData.getLeadCaseId();
+        List<String> caseIds = multipleData.getCaseIdCollection().stream()
+            .map(o -> o.getValue().getEthosCaseReference())
+            .collect(Collectors.toList());
+
+        if (!isNullOrEmpty(leadCaseId)) {
+            caseIds.add(leadCaseId);
+        }
+
+        if (caseIds.isEmpty()) {
+            // No cases linked to this multiple means no legal reps to add
+            return;
+        }
+
+        String token = adminUserService.getAdminUserToken();
+        String multipleCaseTypeId = multipleDetails.getCaseTypeId();
+        String singleCaseTypeId = multipleCaseTypeId.replace("_Multiple", "");
+        
+        List<SubmitEvent> cases = ccdClient.retrieveCasesElasticSearch(token, singleCaseTypeId, caseIds);
+        log.error("Retrieved {} cases from ES", cases.size());
+
+        List<String> orgIds = getUniqueOrganisations(cases);
+
+        List<OrganisationUsersIdamUser> users = orgIds.stream()
+            .map(o -> organisationClient.getOrganisationUsers(token, authTokenGenerator.generate(), o))
+            .flatMap(o -> o.getBody().getUsers().stream())
+            .toList();
+
+        Map<Long, List<String>> emails = getUniqueLegalRepEmails(cases);
+        Map<String, String> legalrepMap = buildEmailIdMap(users);
+
+        HashMap<String, List<String>> legalRepsByCaseId = new HashMap<>();
+
+        for (Entry<Long, List<String>> byCase : emails.entrySet()) {
+            
+            legalRepsByCaseId.put(byCase.getKey().toString(), new ArrayList<>());
+
+            for (String userEmail : byCase.getValue()) {
+                String legalRepId = legalrepMap.get(userEmail);
+
+                legalRepsByCaseId.get(byCase.getKey().toString()).add(legalRepId);
+            }
+        }
+
+        var ethosList = List.of(cases.get(0).getCaseData().getEthosCaseReference());
```

Comment:
Would it be possible to use this list to populate the legalRepCollection on the Multiple? That way the LRs added to Multiple here won't lose access to the Multiple Shell of their case, when a case is transferred between Multiples?

Reply:
I'm not entirely sure I follow here, this ethosList is purely to generate a task for message-handler to deal with. Since msg-handler will process all cases in one go it only needs one case to be present in this list. Related PR: https://github.com/hmcts/et-message-handler/pull/499

Reply:
Sorry, I should have attached this comment to the for loop above. 🤦‍♂️

Would it be possible to construct a second collection that was a list of all the LegalReps grouped by caseEthosRef, and to populate the legalRepCollection of the multiple?

Thanks

Reply:
Check the message-handler PR, it should do this https://github.com/hmcts/et-message-handler/pull/499

Reply:
Thanks I see it now :)